src/submission/exceptions/exceptions.ts

```diff
@@ -260,3 +274,67 @@ export const checkForProgramAndEntityExceptions = async ({
 	});
 	return { filteredErrors, normalizedRecord };
 };
+
+/**
+ * Collect all exception records related to a set of Donors
+ *
+ * @param programId
+ * @param filters
+ * @returns [ ExceptionRecords ]
+ */
+export async function getExceptionManifestRecords(
+	programId: string,
+	filters: { donorIds: number[]; submitterDonorIds: string[] },
+): Promise<ExceptionManifestRecord[]> {
+	const { donorIds, submitterDonorIds: querySubmitterIds } = filters;
+
+	const donorsByDonorId = await getDonorsByIds(donorIds);
+	const donorsBySubmitterId = await Promise.all(
+		querySubmitterIds.map(
+			async (submitterId) => await findDonorBySubmitterId(submitterId, programId),
+		),
+	);
```

Comment:
could add a single repo/mongo/db function somewhere to "find donor by submitter IDs (plural)", then there's no need for mapping through an individual query. possibility to even put some of the filtering there too.

Reply:
Yes! Added. Was surprised such a function didn't exist and now happy it was very easy to create.